jina/clients/__init__.py

```diff
@@ -202,6 +202,53 @@ def reload(
             **kwargs,
         )
 
+    def dump(
+        self,
+        targets: Union[str, List[str]],
+        on_done: CallbackFnType = None,
+        on_error: CallbackFnType = None,
+        on_always: CallbackFnType = None,
+        dump_path: str = None,
```

Comment:
according to signature, dump path and shards should be Optional..

However I think it would be more handy to have them as the first arguments and maybe have no default value